From: Michael Davidsaver <mdavidsaver@bnl.gov>
Date: Wed, 19 Sep 2012 10:14:48 -0400
Subject: cap5: hardcode default base path in perl scripts

Avoid need to play games as install location is known.
---
 src/ca/client/perl/CA.pm         | 5 +----
 src/template/base/makeBaseApp.pl | 5 ++---
 src/template/ext/makeBaseExt.pl  | 3 +--
 3 files changed, 4 insertions(+), 9 deletions(-)

Index: epics-base/src/template/base/makeBaseApp.pl
===================================================================
--- epics-base.orig/src/template/base/makeBaseApp.pl
+++ epics-base/src/template/base/makeBaseApp.pl
@@ -69,7 +69,7 @@ sub ReplaceFilename { # (filename)
             } elsif (-r "$base\@Common") {	# Default version exists
                 $base = '' unless ($filearch eq "Common" && -s $file);
             } else {			# No default version
-                $base = '';
+                $base = '/usr/lib/epics';
             }
             $file = $base;	# Strip the @... part from the target name
         }
@@ -181,8 +181,7 @@ sub get_commandline_opts { #no args
     } elsif ($ENV{EPICS_MBA_BASE}) { # third choice is env var EPICS_MBA_BASE
         $epics_base = UnixPath($ENV{EPICS_MBA_BASE});
     } elsif ($command =~ m|/bin/|) { # assume script was run with full path to base
-        $epics_base = $command;
-        $epics_base =~ s|^(.*)/bin/.*makeBaseApp.*|$1|;
+        $epics_base = '/usr/lib/epics';
     }
     $epics_base and -d "$epics_base" or Cleanup(1, "Can't find EPICS base");
     $app_epics_base = LocalPath($epics_base);
Index: epics-base/src/template/ext/makeBaseExt.pl
===================================================================
--- epics-base.orig/src/template/ext/makeBaseExt.pl
+++ epics-base/src/template/ext/makeBaseExt.pl
@@ -130,8 +130,7 @@ sub get_commandline_opts { #no args
     } elsif ($eBASE) { # third choice is env var EPICS_MBE_BASE
         $epics_base = UnixPath($eBASE);
     } elsif ($command =~ m|/bin/|) { # assume script was called with full path to base
-	$epics_base = $command;
-	$epics_base =~ s|(/.*)/bin/.*makeBaseExt.*|$1|;
+	$epics_base = "/usr/lib/epics";
     }
     "$epics_base" or Cleanup(1, "Cannot find EPICS base");
 
